crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -543,17 +556,55 @@ impl Account {
         Some(key_count as u64)
     }
 
-    pub(crate) fn generate_fallback_key_helper(&mut self) {
-        if self.inner.fallback_key().is_empty() {
+    pub(crate) fn generate_fallback_key_if_needed(&mut self) {
+        if self.inner.fallback_key().is_empty() && self.fallback_key_expired() {
```

Comment:
Could be useful to document that `self.inner.fallback_key().is_empty()` is checking vodozemac's storage, which becomes empty each time `.mark_keys_as_published` is called. So this is checking whether we have a previously generated but non-uploaded fallback key.

Reply:
Added a doc comment: https://github.com/matrix-org/matrix-rust-sdk/pull/3151/commits/f2200b2768cd7d12e5d8d3b14a3b75f4e949c258